pkg/tests/tasks/migration/migration_test.go

```diff
@@ -34,26 +35,49 @@ import (
 	"github.com/maistra/maistra-test-tool/pkg/util/version"
 )
 
+type ingressStatus struct {
+	IP       string `json:"ip,omitempty"`
+	Hostname string `json:"hostname,omitempty"`
+}
+
+func workloadNames(workloads []workload) []string {
+	var names []string
+	for _, wk := range workloads {
+		names = append(names, wk.Name)
+	}
+	return names
+}
+
+type workload struct {
+	Name   string
+	Labels map[string]string
+}
+
+func toSelector(labels map[string]string) string {
+	var parts []string
+	for k, v := range labels {
+		parts = append(parts, k+"="+v)
+	}
+	return strings.Join(parts, ",")
+}
+
 func TestMigrationSimpleClusterWide(t *testing.T) {
 	test.NewTest(t).MinVersion(version.SMCP_2_6).Groups(test.Migration).Run(func(t test.TestHelper) {
 		// delete mesh namespace from previous tests
-		t.LogStepf("Delete namespace %s", meshNamespace)
-		oc.RecreateNamespace(t, meshNamespace)
+		t.LogStep("Cleanup any lingering namespaces from previous runs")
+		oc.DeleteTestBoundNamespaces(t)
+		oc.CreateNamespace(t, meshNamespace)
 
-		istio := ossm.DefaultIstio()
 		t.Cleanup(func() {
-			oc.DeleteResource(t, "", "Istio", istio.Name)
-			oc.DeleteResource(t, "", "IstioCNI", "default")
```

Comment:
why deleting Istio and IstioCNI was removed, Will they be removed after tests?

Reply:
> setupIstio has a cleanUp and namespaces should be deleted by DeleteTestBoundNamespaces IIUC

Yes there's a `t.Cleanup` in `setupIstio` that will remove `Istio` and `IstioCNI` so that you don't have to remember to do it in each one of the tests.

In the same vein, I added a `maistra.io/maistra-test-tool=test-bound-ns` to each one of the namespaces that get created by maistra-test-tool so that for cleanup we can just delete all namespaces with that label rather than having to keep track of the names of each namespace we created.